### DISCUSSION TITLE
MSI WIX version 4 create duplicate inside Program & Features list in Windows 11

### DISCUSSION THREAD

**Discussion by iPk1001101** [wixtoolset/issues#7986] [Questions]
Hello,
In summary, when I use the version 3 I get no issue upgrading the MSI that is already installed. When I try using an MSI generated with the version 4, I get a duplicate in the list of Program & Features of Windows 11. I have gone through the documentation, forums, logs and try to add options but without any luck. Since the post on StackOverflow is quit detailed, I hope it is ok to post it here. I added here the log file of the installation of an MSI made with version 4 to upgrade one made with version 3 here.
https://stackoverflow.com/questions/77957962/wix-msi-version-3-version-4-duplicated-item-in-the-list-of-program-installed
Thanks in advance for any lead on how to fix this! If you want me to redo a complete post here, I will.
[msi-debug-v4.log](https://github.com/wixtoolset/issues/files/14201889/msi-debug-v4.log)

**Comment by nirbar** [comment]
The log says 
```
MSI (c) (50:A4) [17:39:40:464]: Doing action: FindRelatedProducts
MSI (c) (50:A4) [17:39:40:464]: Note: 1: 2205 2:  3: ActionText 
Action 17:39:40: FindRelatedProducts. Searching for related applications Found application: [1]
Action start 17:39:40: FindRelatedProducts.
MSI (c) (50:A4) [17:39:40:465]: FindRelatedProducts: current install is per-machine.  Related install for product '{887DFE8D-6464-4D97-BED3-70E80F38FF67}' is per-user.  Skipping...
MSI (c) (50:A4) [17:39:40:465]: FindRelatedProducts: current install is per-machine.  Related install for product '{887DFE8D-6464-4D97-BED3-70E80F38FF67}' is per-user.  Skipping...
Action ended 17:39:40: FindRelatedProducts. Return value 1.
```
Which means that your package scope has changed from per-user to per-machine

**Reply by iPk1001101** [reply]
Thank you for your input. I tried to add Package/Scope="perUser" and "perUserOrMachine". In both case, the MSI ask for admin elevation (like it does always). However, in the case of this two options I get this error message "The installer has insufficient privileges to access this directory: C\Program Files (x86)\Project. The installation cannot continue. Log on as administrator or contact your system administrator." In the logs, the process is clearly elevated with success (with the local administrator session). I tried "PerMachine" option and the result is the same as the initial issue; duplicate appear in list of Program and Features in Windows 11. Is it possible to do "PerUser" Scope so that the MSI package is upgraded and require elevation in another subsequent opion ? In the package in WIX v.3 there is no definition of the scope but the package install inside location that require admin elevation; it is a "PerUser" but does the job as a "PerMachine" since it elevate and apply changes in the area that are considered "PerMachine".

**Reply by CodeAndWeb** [reply]
I am hitting the same issue. Did you find a solution?
